src/hotspot/share/code/nmethod.cpp

```diff
@@ -1957,7 +1957,9 @@ void nmethod::invalidate_osr_method() {
   }
 }
 
-void nmethod::log_state_change() const {
+void nmethod::log_state_change(const char* reason) const {
+  assert(reason != nullptr, "Must provide a reason");
+
   if (LogCompilation) {
     if (xtty != nullptr) {
       ttyLocker ttyl;  // keep the following output all in one block
```

Comment:
Please, include same info in `LogCompilation` log.

Reply:
Sure, added.